### PR TITLE
Initial build for 12.15.0 ❄️ 

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Upstream: https://inspector.pypi.io/project/azure-storage-file-share/12.15.0/packages/e3/f0/1b28bc315c9af3a793156fdae1e1c6b2bfc66de1882c4b37e5390af7d349/azure-storage-file-share-12.15.0.tar.gz/

Channel: sfe1ed40

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,31 +1,32 @@
...
   host:
     - pip
-    - python >=3.7
+    - python
+    - wheel
+    - setuptools
   run:
-    - python >=3.7
+    - python
     - azure-core >=1.26.0,<2.0.0
```
Comment (danpetry) [inline recipe/meta.yaml]:
```suggestion
    - azure-core >=1.28.0,<2.0.0
```

**Review by danpetry — APPROVED** [review]
Based on suggested change being addressed

**Comment by lorepirri** [comment]
[PKG-3475]

[PKG-3475]: https://anaconda.atlassian.net/browse/PKG-3475?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
     - cryptography >=2.1.4
-    - typing_extensions >=4.0.1
+    - typing-extensions >=4.3.0
     - isodate >=0.6.1
-    - msrest >=0.7.1
```
Comment (lorepirri) [inline recipe/meta.yaml:28]: they have a sort of recipe there, and `msrest` seems to be included:
https://github.com/Azure/azure-sdk-for-python/blob/azure-storage-file-share_12.15.0/conda/conda-recipes/azure-storage/meta.yaml

even though, I see from the `setup.py` that it is not included:
https://github.com/Azure/azure-sdk-for-python/blob/azure-storage-file-share_12.15.0/sdk/storage/azure-storage-file-share/setup.py

**Review by lorepirri — APPROVED** [review]
LGTM, besides others' comments

do not forget to remove the staging channel

File: recipe/meta.yaml (hunk 2 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:28]: The source of truth to me is PyPI and I see no imports of msrest:
```
$ grep msrest -r .
./PKG-INFO:- Removed `msrest` dependency.
./PKG-INFO:- Fixed msrest dependency issue (#16250)
./CHANGELOG.md:- Removed `msrest` dependency.
./CHANGELOG.md:- Fixed msrest dependency issue (#16250)
./azure/storage/fileshare/_generated/_serialization.py:        msrest type and 'key' with the RestAPI encoded key.
./azure/storage/fileshare/_generated/_serialization.py:            # https://github.com/Azure/msrest-for-python/issues/197
./azure/storage/fileshare/_generated/_serialization.py:            # https://github.com/Azure/msrest-for-python/issues/197
./azure_storage_file_share.egg-info/PKG-INFO:- Removed `msrest` dependency.
./azure_storage_file_share.egg-info/PKG-INFO:- Fixed msrest dependency issue (#16250)
```

**Pull request merged**